UPGRADING.md

```diff
@@ -0,0 +1,179 @@
+"babel": {
+  "presets": [
+    "./webpack.babel.js"
+  ]
+}
+```
+
+Finally, download [webpackers babel preset](https://github.com/rails/webpacker/blob/master/package/babel/preset.js) file and place it in the same directory as `packages.json` with the name `webpack.babel.js`.
+
+## 2. Migrate from sass-rails to cssbundling-rails
+
+Start by following these steps:
+
+1. Add `cssbundling-rails` to your Gemfile (do not remove `sass-rails`, Sprockets still needs it);
+2. Run `./bin/bundle install`;
+3. Run `./bin/rails css:install:sass`.
+
+After you are done you will notice that the install step updated some files.
+
+**The new process in 'Procfile.dev'**
+
+Just like the javascript process, this one will bundle and watch for changes in css files.
+
+**The 'build:css' attribute added to packages.json**
+
+This is the command `yarn build` will use to bundle css files.
+
+**The 'link_tree' directive removed from 'app/assets/manifest.js'**
+
+Now that the CSS files will be placed into `app/assets/build`, Sprockets no longer needs to worry about the `app/assets/stylesheets` folder. If you have any other `link_tree` for css files, remove them too.
+
+**Configuring multiple entrypoints**
+
+Sprockets will only compile files in the root directories listed in `manifest.js`, but the sass package that `yarn build` uses will also check subfolders, which might cause compilation errors if your scss files are using features like `@import` and variables. This means that if you have multiple entry points in your app, you have some extra work ahead of you. 
+
+Let's assume you have the following structure in your `app/asset/stylesheets` folder:
+
+```
+stylesheets/admin.scss
+stylesheets/admin/source_1.scss
+stylesheets/admin/source_2.scss
+stylesheets/application.scss
+stylesheets/application/source_1.scss
+stylesheets/application/source_2.scss
+```
+
+Start by your separating your entrypoints from your other files, and adjusting all `@import` for the new structure:
+
+```
+stylesheets/entrypoints/admin.scss
+stylesheets/entrypoints/application.scss
+stylesheets/sources/admin/source_1.scss
+stylesheets/sources/admin/source_2.scss
+stylesheets/sources/application/source_1.scss
+stylesheets/sources/application/source_2.scss
+```
+
+Then adjust the `build` attribute in `packages.json`:
+```
+"build:css": "sass ./app/assets/stylesheets/entrypoints:./app/assets/builds --no-source-map --load-path=node_modules"
+```
+
+**Deprecation warnings**
+
+Sass might raise deprecation warnings depending on what features you are using (such as division), but the messages will explain how to fix them. If you are not sure, see more details in the [official documentation](https://sass-lang.com/documentation/breaking-changes).
+
+## 3. Migrate from Sprockets to Propshaft
+
+Start by following these steps:
+
+1. Replace `sass-rails` with `propshaft`;
+2. Run `./bin/bundle install`;
+3. Open `config/application.rb` and remove `config.assets.paths << Rails.root.join('app','assets')`;
```

Comment:
If you require only the frameworks you need (instead of `rails/all`) you'll also want to remove `require "sprockets/railtie"` here.